src/main/java/org/zalando/nakadi/service/EventPublisher.java

```diff
@@ -114,14 +123,18 @@ private void partition(final List<BatchItem> batch, final EventType eventType) t
     }
 
     private void validate(final List<BatchItem> batch, final EventType eventType) throws EventValidationException,
-            InternalNakadiException {
+            EventSizeValidationException, InternalNakadiException {
         for (final BatchItem item : batch) {
             item.setStep(EventPublishingStep.VALIDATING);
             try {
                 validateSchema(item.getEvent(), eventType);
+                validateEventSize(item);
             } catch (final EventValidationException e) {
                 item.updateStatusAndDetail(EventPublishingStatus.FAILED, e.getMessage());
                 throw e;
+            } catch (final EventSizeValidationException e) {
+                item.updateStatusAndDetail(EventPublishingStatus.ABORTED, e.getMessage());
```

Comment:
I think here it should be `FAILED` like in a case when there was a schema validation problem. Because this is exactly the item that failed.